### PR TITLE
🚀 Early exit for SSR AMP Image 

### PR DESCRIPTION
**Pull request by kristoferbaxter**
Avoid late LCP for attaching the same node again during hydration.

### CODE REVIEW

File: test/unit/test-amp-img.js [hunk 1]
```diff
@@ -572,12 +583,12 @@ describes.sandboxed('amp-img', {}, (env) => {
...
 
     it('does not interfere with SSR img creation', () => {
-      const impl = getImgWithBlur(true, true);
+      const impl = getImgWithBlur(true, true, true);
       const ampImg = impl.element;
-      ampImg.setAttribute('i-amphtml-ssr', '');
       impl.buildCallback();
       impl.layoutCallback();
 
+      // debugger;
```
Comment (kristoferbaxter) [inline test/unit/test-amp-img.js:591]: Oops, left this in.

File: builtins/amp-img.js [hunk 2]
```diff
@@ -191,10 +191,10 @@ export class AmpImg extends BaseElement {
     // fallback to stop from nested fallback abuse.
     this.allowImgLoadFallback_ = !this.element.hasAttribute('fallback');
 
-    // For inabox SSR, image will have been written directly to DOM so no need
-    // to recreate.  Calling appendChild again will have no effect.
+    // For SSR, image will have been written directly to DOM so no need to recreate.
     if (this.element.hasAttribute('i-amphtml-ssr')) {
       this.img_ = scopedQuerySelector(this.element, '> img:not([placeholder])');
+      return this.img_;
```
Comment (jridgewell) [inline builtins/amp-img.js:197]: The SSR can be out of date (and the attributes we currently SSR isn't synchronized with the JS here). If the only LCP issue is reattaching the node, maybe we should just skip the `appendChild` below.

Comment (kristoferbaxter) [reply builtins/amp-img.js:197]: That's a great point, I'll tweak the change.

Comment (kristoferbaxter) [reply builtins/amp-img.js]: Addressed in #782e4cd.

File: test/unit/test-amp-img-intrinsic.js [hunk 3]
```diff
@@ -63,115 +63,89 @@ describes.sandboxed('amp-img layout intrinsic', {}, () => {
...
+        expect(ampImg.getBoundingClientRect()).to.include({
+          width: 100,
+          height: 100,
+        });
+        const img = ampImg.querySelector('img');
+        expect(img.getBoundingClientRect()).to.include({
+          width: 100,
+          height: 100,
+        });
       });
```
Comment (samouri) [inline test/unit/test-amp-img-intrinsic.js:83]: Thank you for improving the test readability!

**Review by samouri — COMMENTED** [review]
Changes LGTM. Do we know the before/after effects already, or do we plan on measuring now that it has landed?

**Pull request merged**